src/command.ts

```diff
@@ -259,8 +273,22 @@ export abstract class Command {
     }
   }
 
+  /**
+   * Determine if the command is being run with the --json flag in a command that supports it.
+   *
+   * @returns {boolean} true if the command supports json and the --json flag is present
+   */
   public jsonEnabled(): boolean {
-    return this.ctor.enableJsonFlag && this.argv.includes('--json')
+    // if the command doesn't support json, return false
+    if (!this.ctor.enableJsonFlag) return false
+    // if the command parameter pass through is enabled, return true if the --json flag is before the '--' separator
+    if (this.passThroughEnabled) {
+      const ptIndex = this.argv.indexOf('--')
```

Comment:
wouldn't this pick up the first `--anything`, like a regular flag ?

Reply:
```suggestion
      const ptIndex = this.argv.indexOf(' -- ')
```

Reply:
@mshanemc no it will not. `this.argv` at this moment is an array of arguments, not a string, so this is finding the index of `--`.

Reply:
yeah, you're right.  It works in all the ways I tried it